tests/Spice86.Tests/Emulator/Devices/Sound/OplGoldenReferenceTests.cs

```diff
@@ -0,0 +1,337 @@
+namespace Spice86.Tests.Emulator.Devices.Sound;
+
+using FluentAssertions;
+using NSubstitute;
+using Spice86.Core.Emulator.CPU;
+using Spice86.Core.Emulator.Devices.ExternalInput;
+using Spice86.Core.Emulator.Devices.Sound;
+using Spice86.Core.Emulator.IOPorts;
+using Spice86.Core.Emulator.VM.Breakpoint;
+using Spice86.Core.Emulator.VM.Clock;
+using Spice86.Core.Emulator.VM.EmulationLoopScheduler;
+using Spice86.Libs.Sound.Common;
+using Spice86.Libs.Sound.Devices.NukedOpl3;
+using Spice86.Shared.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+/// <summary>
+/// Tests comparing Spice86 OPL output against golden reference data from DOSBox Staging.
+/// These tests ensure bit-exact or statistically equivalent audio output for FM synthesis.
+/// </summary>
+public class OplGoldenReferenceTests {
+    private const int OplSampleRateHz = 49716;
+    private const string GoldenReferenceDataDirectory = "Resources/OplGoldenReferences";
+    
+    /// <summary>
+    /// Represents a captured OPL register write sequence.
+    /// Mirrors DOSBox's DRO (DOSBox Raw OPL) format conceptually.
+    /// </summary>
+    private class OplRegisterSequence {
+        public List<OplRegisterWrite> Writes { get; } = new();
+        public int SampleRate { get; set; } = OplSampleRateHz;
+        public string TestName { get; set; } = "unknown";
+        
+        public void AddWrite(ushort port, byte register, byte value, int delayMs = 0) {
+            Writes.Add(new OplRegisterWrite {
+                Port = port,
+                Register = register,
+                Value = value,
+                DelayMs = delayMs
+            });
+        }
+    }
+    
+    /// <summary>
+    /// Represents a single OPL register write with timing.
+    /// </summary>
+    private class OplRegisterWrite {
+        public ushort Port { get; set; }
+        public byte Register { get; set; }
+        public byte Value { get; set; }
+        public int DelayMs { get; set; }
+    }
+    
+    /// <summary>
+    /// Golden reference audio data for comparison.
+    /// </summary>
+    private class GoldenAudioData {
+        public List<AudioFrame> Frames { get; } = new();
+        public int SampleRate { get; set; }
+        public string Source { get; set; } = "DOSBox Staging";
+        
+        public static GoldenAudioData LoadFromFile(string filePath) {
+            if (!File.Exists(filePath)) {
+                return new GoldenAudioData(); // Return empty for missing golden files
+            }
+            
+            GoldenAudioData data = new();
+            // Format: Simple text file with left,right float pairs per line
+            // This is a simplified format; real DOSBox DRO files are binary
```

Comment:
We have dosbox staging source code.

We should be able to have the same capture of FM synth  into DRO files.

And the test should use DRO files, not a different format!